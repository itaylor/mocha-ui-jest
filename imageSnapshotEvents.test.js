const fs = require('fs');
const rimraf = require('rimraf');
const { imageSnapshotSupport } = require('./index.js');
const { makeNewRandomImageBuffer } = require('./pngutils.js');

const { emitter, events, setImageSnapshotConfig } = imageSnapshotSupport;

describe('image snapshot events test', () => {
  let state;
  beforeEach(() => {
    state = wireEvents();
  });
  afterEach(() => emitter.removeAllListeners());

  test('expect works while firing pass events', () => {
    const buf1 = fs.readFileSync(`${__dirname}/fixtures/test-1.png`);
    expect(buf1).toMatchImageSnapshot();
    expect(state).toMatchSnapshot();
  });

  test('expect works while firing fail events', () => {
    expect(() => {
      expect(makeNewRandomImageBuffer()).toMatchImageSnapshot();
    }).toThrow();
    expect(state).toMatchSnapshot();
  });

  test('wont throw when using imageFailuresThrow: false', () => {
    setImageSnapshotConfig({ imageFailuresThrow: false });
    expect(makeNewRandomImageBuffer()).toMatchImageSnapshot();
    expect(state).toMatchSnapshot();
  });

  test('expect works while firing add events', () => {
    rimraf.sync(`${__dirname}/__snapshots__/image-snapshot-events-test-js-expect-works-while-firing-add-events-1-snap.png`);
    expect(makeNewRandomImageBuffer()).toMatchImageSnapshot();
    expect(state).toMatchSnapshot();
    rimraf.sync(`${__dirname}/__snapshots__/image-snapshot-events-test-js-expect-works-while-firing-add-events-1-snap.png`);
  });

  test('expect works while firing update events', () => {
    process.env.SNAPSHOT_UPDATE = 'true';
    expect(makeNewRandomImageBuffer()).toMatchImageSnapshot();
    process.env.SNAPSHOT_UPDATE = 'false';
    expect(state).toMatchSnapshot();
  });
});

function wireEvents() {
  const state = {
    passCount: 0,
    failCount: 0,
    addCount: 0,
    updateCount: 0,
    lastTestContext: null,
  };
  emitter.on(events.IMAGE_SNAPSHOT_PASS, (testContext) => {
    state.lastTestContext = testContext;
    state.passCount++;
  });
  emitter.on(events.IMAGE_SNAPSHOT_FAIL, (testContext) => {
    state.lastTestContext = testContext;
    state.failCount++;
  });
  emitter.on(events.IMAGE_SNAPSHOT_ADD, (testContext) => {
    state.lastTestContext = testContext;
    state.addCount++;
  });
  emitter.on(events.IMAGE_SNAPSHOT_UPDATE, (testContext) => {
    state.lastTestContext = testContext;
    state.updateCount++;
  });
  return state;
}
