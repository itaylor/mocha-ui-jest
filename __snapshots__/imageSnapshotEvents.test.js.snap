// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect works while firing add events 1`] = `
Object {
  "addCount": 1,
  "failCount": 0,
  "lastTestContext": Object {
    "file": "imageSnapshotEvents.test.js",
    "fullTitle": "image snapshot events test expect works while firing add events",
    "snapshotFile": "__snapshots__/image-snapshot-events-test-js-expect-works-while-firing-add-events-1",
    "title": "expect works while firing add events",
  },
  "passCount": 0,
  "updateCount": 0,
}
`;

exports[`expect works while firing fail events 1`] = `
Object {
  "addCount": 0,
  "failCount": 1,
  "lastTestContext": Object {
    "diffOutputPath": "__snapshots__/image-snapshot-events-test-js-expect-works-while-firing-fail-events-1-diff.png",
    "file": "imageSnapshotEvents.test.js",
    "fullTitle": "image snapshot events test expect works while firing fail events",
    "snapshotFile": "__snapshots__/image-snapshot-events-test-js-expect-works-while-firing-fail-events-1",
    "title": "expect works while firing fail events",
  },
  "passCount": 0,
  "updateCount": 0,
}
`;

exports[`expect works while firing pass events 1`] = `
Object {
  "addCount": 0,
  "failCount": 0,
  "lastTestContext": Object {
    "diffOutputPath": "__snapshots__/image-snapshot-events-test-js-expect-works-while-firing-pass-events-1-diff.png",
    "file": "imageSnapshotEvents.test.js",
    "fullTitle": "image snapshot events test expect works while firing pass events",
    "snapshotFile": "__snapshots__/image-snapshot-events-test-js-expect-works-while-firing-pass-events-1",
    "title": "expect works while firing pass events",
  },
  "passCount": 1,
  "updateCount": 0,
}
`;

exports[`expect works while firing update events 1`] = `
Object {
  "addCount": 0,
  "failCount": 0,
  "lastTestContext": Object {
    "file": "imageSnapshotEvents.test.js",
    "fullTitle": "image snapshot events test expect works while firing update events",
    "snapshotFile": "__snapshots__/image-snapshot-events-test-js-expect-works-while-firing-update-events-1",
    "title": "expect works while firing update events",
  },
  "passCount": 0,
  "updateCount": 1,
}
`;

exports[`wont throw when using imageFailuresThrow: false 1`] = `
Object {
  "addCount": 0,
  "failCount": 1,
  "lastTestContext": Object {
    "diffOutputPath": "__snapshots__/image-snapshot-events-test-js-wont-throw-when-using-image-failures-throw--false-1-diff.png",
    "file": "imageSnapshotEvents.test.js",
    "fullTitle": "image snapshot events test wont throw when using imageFailuresThrow: false",
    "snapshotFile": "__snapshots__/image-snapshot-events-test-js-wont-throw-when-using-image-failures-throw--false-1",
    "title": "wont throw when using imageFailuresThrow: false",
  },
  "passCount": 0,
  "updateCount": 0,
}
`;
